### ISSUE TITLE
vkCreateInstance failed -9 invalid gpu device

### ISSUE THREAD

**Issue by t3rminus** [upscayl/upscayl#594]
### Checklist

- [X] I have checked that this issue isn't addressed in [the FAQ](https://github.com/upscayl/upscayl#-faq).
- [X] I have checked that this issue isn't addressed in any other issue or pull request.
- [X] I have tried [setting my GPU ID](https://github.com/upscayl/upscayl/wiki/Guide#gpu-id).
- [ ] I am using Windows and I've tried [common troubleshooting steps](https://github.com/upscayl/upscayl/wiki/Troubleshooting).

### Paste the logs

```sh
⚙️ Getting saveImageAs from localStorage:  jpg
⚙️ Getting model from localStorage:  {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Setting gpuId to empty string
🔀 Setting model to 
🔀 Setting model to realesrgan-x4plus
🐞 BACKEND REPORTED:  💾 Updating Save Output Folder:  false
🐞 BACKEND REPORTED:  📝 Updating Overwrite:  false
🔄 Resetting image paths
⤵️ Dropped file:  {"type":"image/jpeg","filePath":"/home/t3rminus/Downloads/Dusty_2.JPG","extension":"JPG"}
🖼 Setting image path:  /home/t3rminus/Downloads/Dusty_2.JPG
🗂 Setting output path:  /home/t3rminus/Downloads
🖼 imagePath:  /home/t3rminus/Downloads/Dusty_2.JPG
🔤 Extension:  jpg
🔄 Resetting Upscaled Image Path
🏁 UPSCAYL
🐞 BACKEND REPORTED:  📝 Updating Overwrite:  false
🐞 BACKEND REPORTED:  🖼️ Updating No Image Processing:  false
🐞 BACKEND REPORTED:  📐 Updating Compression:  0
🐞 BACKEND REPORTED:  ✅ Upscayl Variables:  {"model":"realesrgan-x4plus","gpuId":null,"saveImageAs":"jpg","inputDir":"/home/t3rminus/Downloads","outputDir":"/home/t3rminus/Downloads","fullfileName":"Dusty_2.JPG","fileName":"Dusty_2","initialScale":"4","desiredScale":"4","outFile":"/home/t3rminus/Downloads/Dusty_2_upscayl_4x_realesrgan-x4plus.jpg","compression":0}
🐞 BACKEND REPORTED:  📢 Upscayl Command:  -i,/home/t3rminus/Downloads/Dusty_2.JPG,-o,/home/t3rminus/Downloads/Dusty_2_upscayl_4x_realesrgan-x4plus.png,-s,4,-m,/usr/lib/upscayl/resources/models,-n,realesrgan-x4plus,,,-f,png
🐞 BACKEND REPORTED:  👶 Updating Child Processes:  {"binary":"/usr/lib/upscayl/resources/bin/upscayl-bin","args":["/usr/lib/upscayl/resources/bin/upscayl-bin","-i","/home/t3rminus/Downloads/Dusty_2.JPG","-o","/home/t3rminus/Downloads/Dusty_2_upscayl_4x_realesrgan-x4plus.png","-s","4","-m","/usr/lib/upscayl/resources/models","-n","realesrgan-x4plus","","","-f","png"]}
🐞 BACKEND REPORTED:  🛑 Updating Stopped:  false
🐞 BACKEND REPORTED:  image upscayl:  vkCreateInstance failed -9
vkCreateInstance failed -9
vkCreateInstance failed -9
invalid gpu device

🔄 Resetting image paths
🚧 UPSCAYL_PROGRESS:  vkCreateInstance failed -9
vkCreateInstance failed -9
vkCreateInstance failed -9
invalid gpu device

🐞 BACKEND REPORTED:  ❌ INVALID GPU OR FAILED
⚙️ Getting saveImageAs from localStorage:  jpg
⚙️ Getting model from localStorage:  {"label":"General Photo (Real-ESRGAN)","value":"realesrgan-x4plus"}
⚙️ Setting gpuId to empty string
⚙️ Getting rememberOutputFolder from localStorage:  false
```

### Describe the Bug

Image upscaling fails because my GPUID is not detected and the steps to enter it manually don't seem to apply-- the GPUID is not listed in the logs.

### To Reproduce

Install from AUR
Attempt to upscale an image

### Version (or commit hash)

2.9.5

### Select your Operating System

Linux

### OS Version

Arch

### GPU Name

Radeon RX 5700

### Expected Behavior

The image is upscaled

### Screenshots

![Screenshot from 2023-12-11 14-14-55](https://github.com/upscayl/upscayl/assets/643473/21d2354e-23e6-486c-ba9f-ac71a4c136a6)

**Comment by NayamAmarshe** [comment]
Seems like a vulkan issue. Try installing mesa packages.

**Comment by t3rminus** [comment]
> Seems like a vulkan issue. Try installing mesa packages.

Mesa was already installed (v23.2.1). Drivers also weren't the issue.

You did, however, point me in the right direction. I'd discounted it being a vulkan issue because I'd been playing Proton/Steam vulkan games without issue... but I realized Steam was running from a flatpak, which undoubtedly bundles the vulkan libraries. **It turns out that vulkan wasn't installed at a system level.**

For anyone else encountering this issue, this page got me going: [https://wiki.archlinux.org/title/Vulkan](https://wiki.archlinux.org/title/Vulkan). The package I needed to install was `vulkan-radeon`.

A list of required libraries/packages might be worth adding to the docs!
